app/src/main/java/com/github/se/travelpouch/ui/notifications/NotificationItem.kt

```diff
@@ -0,0 +1,226 @@
+) {
+  Button(
+      onClick = {
+        handleInvitationResponse(
+            notification,
+            listTravelViewModel,
+            profileViewModel,
+            notificationViewModel,
+            context,
+            isAccepted = false)
+      },
+      colors =
+          ButtonDefaults.buttonColors(containerColor = Color.Transparent, contentColor = Color.Red),
+      modifier = Modifier.testTag("notification_item_decline_button")) {
+        Text(text = "DECLINE")
+      }
+}
+
+fun handleInvitationResponse(
+    notification: Notification,
+    listTravelViewModel: ListTravelViewModel,
+    profileViewModel: ProfileModelView,
+    notificationViewModel: NotificationViewModel,
+    context: android.content.Context,
+    isAccepted: Boolean
+) {
+  listTravelViewModel.getTravelById(
+      notification.travelUid,
+      { travel ->
+        val responseType = if (isAccepted) NotificationType.ACCEPTED else NotificationType.DECLINED
+        val responseMessage = if (isAccepted) "ACCEPTED" else "DECLINED"
+
+        val invitationResponse =
+            Notification(
+                notification.notificationUid,
+                profileViewModel.profile.value.fsUid,
+                notification.senderUid,
+                notification.travelUid,
+                NotificationContent.InvitationResponseNotification(
+                    profileViewModel.profile.value.name, travel!!.title, isAccepted),
+                responseType)
+
+        notificationViewModel.sendNotification(invitationResponse)
+        if (isAccepted) {
+          listTravelViewModel.addUserToTravel(
+              profileViewModel.profile.value.email,
+              travel,
+              { updatedContainer ->
+                listTravelViewModel.selectTravel(updatedContainer)
+                Toast.makeText(context, "User added successfully!", Toast.LENGTH_SHORT).show()
+              },
+              { Toast.makeText(context, "Failed to add user", Toast.LENGTH_SHORT).show() })
+        }
+        Toast.makeText(context, responseMessage, Toast.LENGTH_SHORT).show()
+      },
+      onFailure = { Toast.makeText(context, "Failed to get travel", Toast.LENGTH_SHORT).show() })
+}
+
+fun onCardClick(
+    notification: Notification,
+    listTravelViewModel: ListTravelViewModel,
+    activityViewModel: ActivityViewModel,
+    documentViewModel: DocumentViewModel,
+    eventsViewModel: EventViewModel,
+    navigationActions: NavigationActions,
+    context: android.content.Context
+) {
+  listTravelViewModel.getTravelById(
+      notification.travelUid,
+      onSuccess = { travel ->
+        listTravelViewModel.selectTravel(travel!!)
+        activityViewModel.setIdTravel(travel.fsUid)
+        documentViewModel.setIdTravel(travel.fsUid)
+        eventsViewModel.setIdTravel(travel.fsUid)
+        navigationActions.navigateTo(Screen.TRAVEL_ACTIVITIES)
+      },
+      onFailure = { Toast.makeText(context, "Failed to get travel", Toast.LENGTH_SHORT).show() })
+}
```

Comment:
This fucntion may cause some security issue. If a user receives an invitation notification, does not answer and just click on it, will he be able to access the travel even if he is not a member ?